OLGA/OLGA-Core/src/main/java/semanticstore/ontology/library/generator/generators/SingletonGenerator.java

```diff
@@ -28,6 +28,11 @@
 import java.util.Map;
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 import semanticstore.ontology.library.generator.exceptions.InvalidUriException;
+import semanticstore.ontology.library.generator.generators.AbstractGenerator;
```

Comment:
Does it make sense to make these plugins that are loaded at runtime? It would make it easier for a person adding a new generator...

Reply:
@joephayes 
I actually started using [Dagger](https://dagger.dev/dev-guide/) for dependency injection as you are pointing out.
Then I realized others might not use it to extend OLGA.
I think it makes sense to seperate them and run the tests of code generation for all three.
But the `compilation` tests we can't assert unless the underlying dependencies are installed.

thoughts?

Reply:
@charbull Sorry for the delay!

At a high level, I'm thinking in terms of modules (plugins). OLGA would maintain a registry of modules and load the code as needed. 

What if OLGA defined these as service interfaces and then the Python, C# and Java generators can be implemented as services? OLGA could then use the Java [ServiceLoader](https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/util/ServiceLoader.html) to load them.

If OLGA adopted this model, the code for the different generators would move into separate repos and the compilation tests would move as well.

It would also give a definitive answer to the question "how do I add a new generator?"

Reply:
@charbull - is this something you want to discuss more? Maybe I should open a separate issue?

Reply:
@joephayes I think it makes sense, I will start looking at it.
For the RDF python generator do you have some bandwidth to collaborate in co-development?

Reply:
@charbull - A colleague of mine and I will have some bandwidth towards the end of the month and we could work on the generator for RDF Python then.

Reply:
That's great ! 
Will send another PR for the service loader

Reply:
Sounds great! Looking forward to it!